js/widgets.js

```diff
@@ -1,108 +1,172 @@
-const datasets = [].slice.call(document.querySelectorAll('[data-udata-dataset-id]'));
-// Extract the base URL from the URL to that script.
-const parser = document.createElement('a');
-const scriptURI = document.querySelector('script#udata').src
-parser.href = scriptURI;
-const baseURL = `${parser.protocol}//${parser.host}`;
-// Keep the chunk > 9 otherwise territories pages will issue more than one query.
-territories.concat(datasets).chunk(9).forEach((elements) => {
+const embedDatasets = (territories, datasets) => {
+  // Main function retrieving the HTML code from the API.
+  // Keep the chunk > 12 otherwise territories pages will issue more than one query.
+  chunk(territories.concat(datasets), 12).forEach((elements) => {
     const references = elements.map((el) => {
-        if (el.hasAttribute('data-udata-territory-id')) {
-            const territoryId = el.getAttribute('data-udata-territory-id');
-            return `territory-${territoryId}`;
-        } else if (el.hasAttribute('data-udata-dataset-id')) {
-            const datasetId = el.getAttribute('data-udata-dataset-id');
-            return `dataset-${datasetId}`;
-        }
-    });
-    const url = `${baseURL}/api/1/oembeds/?references=${references}`;
+      if (el.hasAttribute(TERRITORY_ID)) {
+        return `territory-${el.dataset.udataTerritoryId}`
+      } else if (el.hasAttribute(DATASET_ID)) {
+        return `dataset-${el.dataset.udataDatasetId}`
+      }
+    })
+    const url = `${baseURL}/api/1/oembeds/?references=${references}`
+    // Polyfill added by webpack.
     fetch(url)
-        .then(checkStatus)
-        .then((response) => response.json())
-        .then((jsonResponse) => {
-            // We match the returned list with the list of elements.
-            zip([elements, jsonResponse, references]).forEach(([element, response, id]) => {
-                element.innerHTML = response.html;
-                document.querySelector(`#${id} .integrate`).addEventListener('click', handleIntegration)
-            });
+      .then(checkStatus)
+      .then((response) => response.json())
+      .then((jsonResponse) => {
+        // We match the returned list with the list of elements.
+        zip([elements, jsonResponse, references])
+          .forEach(([element, response, id]) => {
+            element.innerHTML = response.html
+            Promise.resolve()
+              .then(() => {
+                const integrateElement = qS(`#${id} .integrate`)
+                if (integrateElement) {
+                  integrateElement.addEventListener('click', handleIntegration)
+                }
+              })
+          })
+      })
+      .catch(console.error.bind(console))
+  })
+}
+
+// If a whole territory is defined, it takes precedence.
+// Otherwise fallback on ids.
+const territoryElement = qS(`[${TERRITORY}]`)
+if (territoryElement) {
+  const territorySlug = territoryElement.dataset.udataTerritory
```

Comment:
OMG !!! Is there black magic or is the dash to camel native ?
If it's the case, good to know (and then you can safely ignore the previous comment on the subject)